sync3/room.go

```diff
@@ -22,7 +22,19 @@ type Room struct {
 	NumLive           int               `json:"num_live,omitempty"`
 }
 
+// RoomConnMetadata represents a room as seen by one specific connection (hence once
+// specific device).
 type RoomConnMetadata struct {
+	// We enclose copies of the data kept in the global and user caches. These snapshots
+	// represent the state we reported to the connection the last time they requested
+	// a sync. Note that we are free to tweak fields within these copies if we want, to
+	// report more appropriate data to clients that kept in the caches.
```

Comment:
Wording is off here.

On tweaking fields: the only examples I could see of this today were

- the LastMessageTimestamp, which I explicitly want to pull out to a per-list map on the RoomConnMetadata; and
- the CanonicalisedName, which I'm suspicious of. I don't really understand why it needs to be computed in SetRoom.

Reply:
I understand why it's computed eagerly; I don't understand why we compute it eagerly per-conn, rather than in the user cache itself.

Reply:
I.e. why would one conn have a different CanonicalisedName compared to any other for the same user, other than the conn being "behind" the latest data in the user cache?